webservice/src/views/ProfessorView.jsx

```diff
@@ -1,29 +1,178 @@
 import React from 'react';
 import TableRow from '@material-ui/core/TableRow';
-import LabTemplatesList from '../components/LabTemplatesList';
-import LabInstancesList from '../components/LabInstancesList';
+import Grid from '@material-ui/core/Grid';
+import Slide from '@material-ui/core/Slide';
+import Button from '@material-ui/core/Button';
+import AddCircleIcon from '@material-ui/icons/AddCircle';
+import Dialog from '@material-ui/core/Dialog';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogActions from '@material-ui/core/DialogActions';
+import CloudUploadIcon from '@material-ui/icons/AddCircle';
+import DeleteIcon from '@material-ui/icons/Cancel';
+import TemplateForm from '../components/NewTemplateForm';
 import { labPapersStyle } from './StudentView';
-import ListSubheader from '@material-ui/core/ListSubheader';
+import LabInstancesList from '../components/LabInstancesList';
+import LabTemplatesList from '../components/LabTemplatesList';
 
 export default function ProfessorView(props) {
   return (
     <>
       <TableRow style={labPapersStyle}>
         <LabTemplatesList
+          delete={props.delete}
           labs={props.templateLabs}
           func={props.funcTemplate}
           start={props.start}
-          isAdmin={true}
+          isAdmin
         />
         <LabInstancesList
           runningLabs={props.instanceLabs}
           func={props.funcInstance}
           connect={props.connect}
           stop={props.stop}
           showStatus={props.showStatus}
-          isAdmin={true}
+          isAdmin
         />
       </TableRow>
+      <TableRow style={labPapersStyle}>
+        <Grid
+          container
+          spacing={0}
+          alignItems="center"
+          justify="center"
+          direction="row"
+          noValidate
+          autoComplete="off"
+        >
+          <NewTemplateWrapper
+            registryName={props.registryName}
+            imageList={props.imageList}
+            funcNewTemplate={props.funcNewTemplate}
+            adminGroups={props.adminGroups}
+          />
+        </Grid>
+      </TableRow>
     </>
   );
 }
+
+const Transition = React.forwardRef(function Transition(props, ref) {
+  return <Slide direction="up" ref={ref} {...props} />;
+});
+
+const NewTemplateWrapper = props => {
+  const [open, setOpen] = React.useState(false);
+  const [image, setImage] = React.useState(null);
+  const [version, setVersion] = React.useState(null);
+  const [labid, setLabid] = React.useState(null);
+  const [namespace, setNamespace] = React.useState(null);
+  const [errorcode, setErrorcode] = React.useState(0);
+
+  const handleClickOpen = () => {
+    setOpen(true);
+  };
+  const handleClose = () => {
+    if (!props.adminGroups.includes(namespace)) {
+      setErrorcode(1);
+      return;
+    }
+    if (labid === null) {
```

Comment:
Here, you should also check whether the textbox is empty. Otherwise, if you type something and then delete it, no error is raised and an incorrect labtemplate is created.